File: scripts/generate-screenshot.js
#!/usr/bin/env node

const puppeteer = require('puppeteer');
const fs = require('fs');
const path = require('path');

async function generateScreenshots() {
    console.log('📸 Generating screenshots...');

    const htmlFile = path.join(__dirname, '../data/index.html');
    const screenshotDir = path.join(__dirname, '../data/screenshots');

    // Create screenshots directory
    if (!fs.existsSync(screenshotDir)) {
        fs.mkdirSync(screenshotDir, { recursive: true });
    }

    const browser = await puppeteer.launch({
        headless: 'new',
        args: ['--no-sandbox', '--disable-setuid-sandbox']
    });

    try {
        const page = await browser.newPage();

        // Load HTML file
        const htmlContent = fs.readFileSync(htmlFile, 'utf8');
        await page.setContent(htmlContent, { waitUntil: 'networkidle0' });

        // Wait a bit for animations
        await page.waitForTimeout(3000);

        // 1. Full page screenshot for documentation
        console.log('   Generating full page screenshot...');
        await page.screenshot({
            path: path.join(screenshotDir, 'stats-full.png'),
            fullPage: true,
            type: 'png',
            quality: 90
        });

        // 2. Optimized screenshot for social media (Twitter/Facebook)
        console.log('   Generating social media preview...');
        await page.setViewport({ width: 1200, height: 630 });
        await page.waitForTimeout(500);
        await page.screenshot({
            path: path.join(screenshotDir, 'stats-social.png'),
            type: 'png',
            quality: 90
        });

        // 3. Screenshot for GitHub README (optimized width)
        console.log('   Generating README screenshot...');
        await page.setViewport({ width: 1280, height: 720 });
        await page.waitForTimeout(500);
        await page.screenshot({
            path: path.join(screenshotDir, 'stats-readme.png'),
            type: 'png',
            quality: 90
        });

        console.log('✅ Screenshots generated successfully!');
        console.log(`📁 Location: ${screenshotDir}`);

    } catch (error) {
        console.error('❌ Error generating screenshots:', error);
        throw error;
    } finally {
        await browser.close();
    }
}

// Run if called directly
if (require.main === module) {
    generateScreenshots().catch(error => {
        console.error(error);
        process.exit(1);
    });
}

module.exports = generateScreenshots;

File: scripts/generate-screenshot.py
#!/usr/bin/env python3
"""
Generate screenshots using imgkit (wkhtmltoimage)
Lightweight and fast
"""

import os
import sys
from pathlib import Path

# Try to import imgkit
try:
    import imgkit
except ImportError:
    print("Installing imgkit...")
    os.system(f"{sys.executable} -m pip install imgkit")
    import imgkit

def generate_screenshots():
    print("📸 Generating screenshots...")
    
    # Setup paths
    base_dir = Path(__file__).parent.parent
    html_file = base_dir / 'data' / 'index.html'
    screenshot_dir = base_dir / 'data' / 'screenshots'
    
    # Create output directory
    screenshot_dir.mkdir(parents=True, exist_ok=True)
    
    if not html_file.exists():
        print(f"❌ HTML file not found: {html_file}")
        return False
    
    # Configure imgkit options
    options = {
        'format': 'png',
        'quality': 90,
        'encoding': "UTF-8",
        'enable-local-file-access': None,
        'quiet': ''
    }
    
    try:
        print("   Generating full page screenshot...")
        imgkit.from_file(
            str(html_file),
            str(screenshot_dir / 'stats-full.png'),
            options=options
        )
        
        print("   Generating social media preview...")
        custom_options = options.copy()
        custom_options.update({'width': 1200, 'height': 630})
        imgkit.from_file(
            str(html_file),
            str(screenshot_dir / 'stats-social.png'),
            options=custom_options
        )
        
        print("   Generating README screenshot...")
        custom_options = options.copy()
        custom_options.update({'width': 1280, 'height': 720})
        imgkit.from_file(
            str(html_file),
            str(screenshot_dir / 'stats-readme.png'),
            options=custom_options
        )
        
        print(f"✅ Screenshots saved to: {screenshot_dir}")
        return True
        
    except Exception as e:
        print(f"❌ Error generating screenshots: {e}")
        return False

if __name__ == "__main__":
    success = generate_screenshots()
    sys.exit(0 if success else 1)